data/transactions/application.go

```diff
@@ -620,7 +621,12 @@ func (ac *ApplicationCallTxnFields) apply(header Header, balances Balances, spec
 	// ApprovalProgram, since clearing out is always allowed. We only
 	// execute the ClearStateProgram, whose failures are ignored.
 	if ac.OnCompletion == ClearStateOC {
-		return ac.applyClearState(balances, header.Sender, appIdx, ad, steva)
+		err = ac.applyClearState(balances, header.Sender, appIdx, ad, steva)
+		if err != nil {
```

Comment:
I hate to bikeshed too much on this, but I think this logic would be better at the end of `applyClearState` instead of here. Otherwise we're setting `ad.EvalDelta = basics.EvalDelta{}` in two different methods instead of just one, which makes the idea less clear. Alternatively, I would be ok with leaving this here and deleting line 526.

Reply:
completely makes sense, done.